### PR TITLE
Old routers may not have permission to do SAR checks for metrics

### PR DESCRIPTION
**Pull request by smarterclayton**
Fallback to the old unauthorized status code. Do the same for the
authentication and authorization checks, although authorization should
be forbidden.

Contains #17259

### CODE REVIEW

**Comment by smarterclayton** [comment]
@stevekuznetsov

File: pkg/router/metrics/metrics.go [hunk 1]
```diff
@@ -62,9 +63,16 @@ func (l Listener) authorizeHandler(protected http.Handler) http.Handler {
...
 
 		user, ok, err := l.Authenticator.AuthenticateRequest(req)
-		if err != nil {
-			glog.V(3).Infof("Unable to authenticate: %v", err)
-			http.Error(w, "Unable to authenticate due to an error", http.StatusInternalServerError)
+		if !ok || err != nil {
+			// older routers will not have permission to check token access review, so treat this
+			// as an authorization denied if so
+			if !ok || errors.IsUnauthorized(err) {
+				glog.V(5).Infof("Unable to authenticate: %v", err)
```
Comment (rajatchopra) [inline pkg/router/metrics/metrics.go]: It might show nothing in case of err==nil and !ok, but it's harmless.

File: pkg/router/metrics/metrics.go [hunk 2]
```diff
@@ -62,9 +63,16 @@ func (l Listener) authorizeHandler(protected http.Handler) http.Handler {
...
+				http.Error(w, "Unable to authenticate due to an error", http.StatusUnauthorized)
```
Comment (knobunc) [inline pkg/router/metrics/metrics.go]: Do we want to show that string to the end user?

**Comment by stevekuznetsov** [comment]
Going to manually merge this one to unblock us in the AMI.

**Pull request merged**